Source/WebCore/html/CheckboxInputType.h

```diff
@@ -57,13 +57,23 @@ class CheckboxInputType final : public BaseCheckableInputType {
     String valueMissingText() const final;
     void createShadowSubtree() final;
     void handleKeyupEvent(KeyboardEvent&) final;
+    void handleMouseDownEvent(MouseEvent&) final;
+    void handleMouseMoveEvent(MouseEvent&) final;
+    void startSwitchPointerTracking(int);
+    void stopSwitchPointerTracking();
+    bool isSwitchPointerTracking() const;
     void willDispatchClick(InputElementClickState&) final;
     void didDispatchClick(Event&, const InputElementClickState&) final;
     bool matchesIndeterminatePseudoClass() const final;
+    void willUpdateCheckedness(bool /* nowChecked */, WasSetByJavaScript);
     void disabledStateChanged() final;
+    void performSwitchCheckedChangeAnimation();
     void stopSwitchCheckedChangeAnimation();
     void switchCheckedChangeAnimationTimerFired();
 
+    int m_switchPointerTrackingXPositionStart { -1 };
+    bool m_hasSwitchVisuallyOnChanged { false };
+    bool m_isSwitchVisuallyOn { false };
```

Comment:
In the future, we should probably consider storing switch-specific data as a sidecar object, since most checkboxes don't need this information.